File: Practica12/EJ1/Java/Patron.java
public class Patron {
    private static Patron instancia;
    private Patron() {
    }
    public static Patron getInstancia() {
        if (instancia == null) {
            instancia = new Patron();
        }
        return instancia;
    }
    public void operacion() {
        System.out.println("ejecutando operación del Patron");
    }
    public static void main(String[] args) {
        Patron s1 = Patron.getInstancia();
        Patron s2 = Patron.getInstancia();
        s1.operacion();
        System.out.println(s1 == s2);
    }
}

File: Practica12/EJ1/Python/Patron.py
class Singleton:
    _instancia = None
    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls)
        return cls._instancia
    def operacion(self):
        print("Ejecutando operación del Singleton.")
s1 = Singleton()
s2 = Singleton()
s1.operacion()
print(s1 is s2)
